pkg/controller/collector/timestamp.go

```diff
@@ -57,43 +65,41 @@ func NewTimestamp(config TimestampConfig) (*Timestamp, error) {
 }
 
 func (t *Timestamp) Collect(ch chan<- prometheus.Metric) error {
-	ctx := context.Background()
-	gvk, err := apiutil.GVKForObject(t.newRuntimeObjectFunc(), t.k8sClient.Scheme())
-	if err != nil {
-		return microerror.Mask(err)
+	list := &metav1.PartialObjectMetadataList{}
```

Comment:
This type allows us to only get metadata rather than the whole object, saving memory and network traffic.